addons/web/static/src/core/domain_selector/domain_selector.js

```diff
@@ -9,6 +9,7 @@ import {
     buildDomain,
     buildDomainSelectorTree,
     extractPathsFromDomain,
+    formatValueFromFieldType,
```

Comment:
the code seems good.
For the commit message, I suggest smth like that

[FIX] web: localize date/datetime in domain descriptions

The date/datetime values produced in the domain selector for date/datetime fields are not localized (like they are in the date picker) when presented in

- the domain selector in readonly mode
- the facets edited/created from the domain selector dialog

Here we make sure that the localization parameters are used when we display date/datetime values in the domain descriptions.